File: dotnet/xjustiz.core-dotnet/Models/Datei.cs
namespace xjustiz.core_dotnet.Models;

using System.Collections.Generic;
using System.Xml.Serialization;

/// <summary>
/// Repräsentiert eine Datei.
/// Represents a file.
/// </summary>
public class Datei
{
    /// <summary>
    /// Der Dateiname.
    /// The filename.
    /// </summary>
    [XmlElement("dateiname", Namespace = XJustizExport.Tns, Order = 1)]
    public string? Dateiname { get; set; }

    /// <summary>
    /// Bestandteile der Datei.
    /// Components of the file.
    /// </summary>
    [XmlElement("bestandteil", Namespace = XJustizExport.Tns, Order = 2)]
    public List<Bestandteil> Bestandteile { get; set; } = new();
}

File: java/src/main/java/de/xjustiz/core/models/Datei.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlElement;
import org.jetbrains.annotations.Nullable;

/**
 * Repräsentiert eine Datei.<br/>
/// <u><b>File:</b></u> Represents a file.
 */
@XmlAccessorType(XmlAccessType.FIELD)
public class Datei {

    @XmlElement(name = "dateiname", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "dateiname", namespace = "http://www.xjustiz.de")
    @JsonProperty("Dateiname")
    @Nullable
    private String dateiname;

    @XmlElement(name = "bestandteil", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "bestandteil", namespace = "http://www.xjustiz.de")
    @JsonProperty("Bestandteil")
    private Bestandteiltyp bestandteil;

    public Datei() {
    }

    @Nullable
    public String getDateiname() {
        return dateiname;
    }

    public void setDateiname(@Nullable String dateiname) {
        this.dateiname = dateiname;
    }

    public Bestandteiltyp getBestandteil() {
        return bestandteil;
    }

    public void setBestandteil(Bestandteiltyp bestandteil) {
        this.bestandteil = bestandteil;
    }

}
